kani-driver/src/call_goto_instrument.rs

```diff
@@ -160,6 +162,30 @@ impl KaniSession {
         self.call_goto_instrument(args)
     }
 
+    /// Make CBMC enforce a function contract.
+    pub fn instrument_contracts(&self, harness: &HarnessMetadata, file: &Path) -> Result<()> {
```

Comment:
Can you please improve the structure of this function? You have an early return if `check` is `None`. Then there's some code that runs inside a `if let Some(assigns) = check`, but not others, such as the call to `goto_instrument`.

Instead, you should wrap the entire code after line 167 inside the `if let Some(assigns) = check {}` block.

Reply:
Good point. I kept the early return but simplified the latter half significantly

Reply:
That looks great. Thanks